lib/models/cql_calculator.js

```diff
@@ -0,0 +1,643 @@
+/**
+ * The CQL calculator. This calls the cql-execution framework and formats the results as neccesary for Bonnie
+ * and Cypress.
+ */
+
+const _ = require('lodash');
+const moment = require('moment');
+const CqmModels = require('cqm-models');
+const Cql = require('cqm-models').CQL;
+const Measure = require('cqm-models').Measure();
+
+module.exports = class CQLCalculator {
+  constructor() {
+    this.calculator = {};
+    this.calculatorLoaded = {};
+    this.resultsCache = {};
+    this.valueSetsByMongoidCached = {};
+  }
+
+  // Key for storing calculators on a measure
+  static calculationKey(measure) { return `${measure.id}`; }
+
+  // Key for storing results for a patient / population calculation
+  // we use the mongo ID for the patient portion
+  cacheKey(measure, patient) { return `${this.calculationKey(measure)}/${patient.id()}`; }
+
+  clearResult(measure, patient) {
```

Comment:
I don't believe `calculationKey`, `cacheKey`, or `clearResult` are ever used. Unless there is a plan to use them in the future they should be removed and if there is a plan to use them in the future, add a comment explaining that.